### PR TITLE
[libigl]Add port libigl, add features.

### PR DESCRIPTION
**Pull request by JackBoosY**
This port does not support arm / uwp.
Related: #2601.

### CODE REVIEW

**Comment by xarthurx** [comment]
Finally this lib has been added~
Thanks~

**Comment by xarthurx** [comment]
When will this be merged....

**Review by PhoebeHui — APPROVED** [review]

File: ports/libigl/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+		-DPYTHON_EXECUTABLE=${PYTHON2}
+)
+
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/libigl/cmake)
+vcpkg_copy_pdbs()
+
+if (NOT LIBIGL_BUILD_STATIC)
+    # For dynamic build, libigl is a header-only library.
```
[the remark this answers is not in the document]
Comment (JackBoosY) [reply ports/libigl/portfile.cmake]: In dynamic mode, libigl's `CMakeLists.txt` will copy headers and do other things,

**Comment by JackBoosY** [comment]
@vicroms ready to merge now.

**Comment by xarthurx** [comment]
Thanks to the quick response, but got an issue:
https://github.com/microsoft/vcpkg/issues/8811

**Pull request merged**